.gitlab-ci.yml

```diff
@@ -50,13 +65,22 @@ stages:
     - sed -i "s#<NFS_PATH>#${NFS_PATH}#g" deployment.yaml
     - kustomize edit set image DOCKER_IMAGE=${DOCKER_IMAGE}
     - kubectl apply -k ./
-  rules:
-    - if: '$CI_DEPLOY_FREEZE == null && $CI_COMMIT_BRANCH == "main"'
-  when: manual
 
-# Run Tests
+.deploy-dev:
+  extends:
+    - .deploy
+    - .branch-rules-dev
+
+.deploy-main:
+  extends:
+    - .deploy
+    - .branch-rules-main
+
+# Run tests
 Test:
   stage: test
+  extends:
+    - .deploy-rules
```

Comment:
I am probably missing something; but why should the Test job be concerned with whether or not something needs to be deployed (and whether or not there is a deploy freeze)?

Reply:
It depends whether we want to run the code tests when we know it's not going to be deployed downstream.

Reply:
I would have thought we want to run tests on every push regardless of where it goes to get early feedback. At least this is how it is for the client.